app/Http/Search/Request.php

```diff
@@ -692,13 +710,62 @@ public function addKnnAndRankParam($params, $input)
                         'num_candidates' => 50,
                     ]
                 ],
-                'weight' => 1.0
+                'weight' => $this->vectorWeight,
             ];
         }
 
         return $params;
     }
 
+    /**
+     * Add param for vector search
+     *
+     * @link https://www.elastic.co/guide/en/elasticsearch/reference/8.18/query-dsl-script-score-query.html#vector-functions-cosine
+     *
+     * @param array $params
+     *
+     * @return array
+     */
+    public function adjustRetrieverWeights($params, $input)
```

Comment:
Will you type hint the param / return value as noted in the method comment block.

Reply:
Good call. There are a number of methods that don't have it, I'll add them.